### PR TITLE
Update script to support secure MQTT broker

### PR DESCRIPTION
**Pull request by cherrycl**
1. Use external mqtt broker on export tests
2. Get credential from vault to subscribe internal secure mqtt message bus.
 
Signed-off-by: Cherry Wang <cherry@iotechsys.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [X] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: TAF/testScenarios/integrationTest/UC_app_service_configurable/external_mqtt_trigger.robot [hunk 1]
```diff
@@ -15,29 +15,22 @@ ExternalTrigger001 - Test external mqtt trigger works
     Given Set Test Variable  ${publish_msg}  Message from mqtt publisher
     And Set Test Variable  ${mqtt_trigger_topic}  external-request
     And Set Test Variable  ${mqtt_export_topic}  edgex-export
-    And Subscribe MQTT Broker Topics
-    When Run process  python ${WORK_DIR}/TAF/utils/src/setup/mqtt-publisher.py ${mqtt_trigger_topic} "${publish_msg}"
+    And Subscribe MQTT Broker Topics ${mqtt_trigger_topic} With ${BROKER_PORT}  # Subscribe trigger mqtt broker
```
Comment (lenny-intel) [inline TAF/testScenarios/integrationTest/UC_app_service_configurable/external_mqtt_trigger.robot:18]: Trigger comes from external broker
```suggestion
    And Subscribe MQTT Broker Topics ${mqtt_trigger_topic} With ${EX_BROKER_PORT}  # Subscribe trigger mqtt broker
```

File: TAF/testScenarios/integrationTest/UC_app_service_configurable/external_mqtt_trigger.robot [hunk 2]
```diff
@@ -15,29 +15,22 @@ ExternalTrigger001 - Test external mqtt trigger works
...
+    And Subscribe MQTT Broker Topics ${mqtt_export_topic} With ${EX_BROKER_PORT}  # Subscribe export mqtt broker
+    When Run process  python ${WORK_DIR}/TAF/utils/src/setup/mqtt-publisher.py ${mqtt_trigger_topic} "${publish_msg}" ${BROKER_PORT}
```
Comment (lenny-intel) [inline TAF/testScenarios/integrationTest/UC_app_service_configurable/external_mqtt_trigger.robot:20]: Publish to external broker for the trigger

```suggestion
    When Run process  python ${WORK_DIR}/TAF/utils/src/setup/mqtt-publisher.py ${mqtt_trigger_topic} "${publish_msg}" ${EX_BROKER_PORT}
```

File: TAF/testScenarios/integrationTest/UC_kuiper/export_by_kuiper_rules.robot [hunk 3]
```diff
@@ -25,9 +25,9 @@ Kuiper001 - Add a new rule and export to MQTT
     Given Set Test Variable  ${device_name}  kuiper-mqtt-device
     And Set Test Variable  ${command}  ${PREFIX}_GenerateDeviceValue_INT8_RW
     And Set Test Variable  ${resource}  ${PREFIX}_DeviceValue_INT8_RW
-    And Start process  python ${WORK_DIR}/TAF/utils/src/setup/mqtt-subscriber.py rules-events-kuiper ${resource} arg &   # Process for MQTT Subscriber
+    And Start process  python ${WORK_DIR}/TAF/utils/src/setup/mqtt-subscriber.py rules-events-kuiper ${resource} ${EX_BROKER_PORT} false arg &   # Process for MQTT Subscriber
```
Comment (lenny-intel) [inline TAF/testScenarios/integrationTest/UC_kuiper/export_by_kuiper_rules.robot:28]: Is this expecting a rule action in eKuiper to export data to the external broker?

File: TAF/utils/scripts/docker/get-compose-file.sh [hunk 4]
```diff
@@ -41,7 +41,9 @@ for compose in ${COMPOSE_FILE}; do
 
   sed -i '/PROFILE_VOLUME_PLACE_HOLDER: {}/d' ${compose}.yml
   sed -i 's/\EXPORT_HOST_PLACE_HOLDER/${DOCKER_HOST_IP}/g' ${compose}.yml
-  sed -i 's/\MQTT_BROKER_ADDRESS_PLACE_HOLDER/${MQTT_BROKER_HOSTNAME}/g' ${compose}.yml
+  sed -i 's/TRIGGER_EXTERNALMQTT_URL: MQTT_BROKER_ADDRESS_PLACE_HOLDER/TRIGGER_EXTERNALMQTT_URL: tcp:\/\/${MQTT_BROKER_HOSTNAME}:1883/g' ${compose}.yml
```
Comment (lenny-intel) [inline TAF/utils/scripts/docker/get-compose-file.sh:44]: The trigger should come from the external broker
```suggestion
  sed -i 's/TRIGGER_EXTERNALMQTT_URL: MQTT_BROKER_ADDRESS_PLACE_HOLDER/TRIGGER_EXTERNALMQTT_URL: tcp:\/\/${EXTERNAL_BROKER_HOSTNAME}:1883/g' ${compose}.yml
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: TAF/testScenarios/integrationTest/UC_kuiper/export_by_kuiper_rules.robot (hunk 3 above)
Comment (cherrycl) [reply TAF/testScenarios/integrationTest/UC_kuiper/export_by_kuiper_rules.robot:28]: Yes, this is expecting to export data to the external broker. Would you expect the action to export data to the internal broker?

File: TAF/testScenarios/integrationTest/UC_app_service_configurable/external_mqtt_trigger.robot (hunk 1 above)
Comment (cherrycl) [reply TAF/testScenarios/integrationTest/UC_app_service_configurable/external_mqtt_trigger.robot]: Updated.

File: TAF/testScenarios/integrationTest/UC_app_service_configurable/external_mqtt_trigger.robot (hunk 2 above)
Comment (cherrycl) [reply TAF/testScenarios/integrationTest/UC_app_service_configurable/external_mqtt_trigger.robot]: Updated

File: TAF/utils/scripts/docker/get-compose-file.sh (hunk 4 above)
Comment (cherrycl) [reply TAF/utils/scripts/docker/get-compose-file.sh]: Updated. Replace all `MQTT_BROKER_ADDRESS_PLACE_HOLDER` to `tcp:\/\/${EXTERNAL_BROKER_HOSTNAME}:1883`

File: TAF/testScenarios/integrationTest/UC_kuiper/export_by_kuiper_rules.robot (hunk 3 above)
Comment (lenny-intel) [reply TAF/testScenarios/integrationTest/UC_kuiper/export_by_kuiper_rules.robot:28]: Nope, just clarifying. :-)

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**